rclcpp/src/rclcpp/utilities.cpp

```diff
@@ -84,9 +84,9 @@ rclcpp::remove_ros_arguments(int argc, char const * const argv[])
     &nonros_argc,
     &nonros_argv);
 
-  if (RCL_RET_OK != ret) {
+  if (RCL_RET_OK != ret || nonros_argc < 0) {
```

Comment:
I think the original `RCL_RET_OK != ret` check is enough. It looks like the only case where it could be negative in `rcl_remove_ros_arguments` already returns `RCL_RET_INVALID_ARGUMENT`

https://github.com/ros2/rcl/blob/d94c958a23308299fccb9398153016fc64d81bd3/rcl/src/rcl/arguments.c#L449-L454

Reply:
its true, that when you open up the implementation you can trace back, that the argc will never be smaller then 0 and the RCL_RET_OK. However the implementation may change without the client library getting notice of it. And if it ever returns a negative value, dire things will happen: a -1 will be std::numeric_limits<size_t>::max(), which will lead to the allocation of a very large data chunk. What do you think?